src/translation.cpp

```diff
@@ -51,63 +62,45 @@ const std::wstring &Translations::getTranslation(
 const std::wstring &Translations::getPluralTranslation(
 		const std::wstring &textdomain, const std::wstring &s, unsigned long int number) const
 {
-	std::wstring key = textdomain + L"|"; key.append(s);
-
-	/*~ DO NOT TRANSLATE THIS LITERALLY!
-	This is a special string which needs to be translated to the translation plural identifier.
-
-	If you edit the po file directly, you should see the following:
-
-		msgid "0"
-		msgid_plural "1"
-		msgstr[0] ...
-		...
-		msgstr[n] ...
-
-	You then need to translate msgstr[i] to the string containing the integer i, like this:
-
-		msgstr[0] "0"
-
-
-	If you edit the po file using weblate or another po file editor, the translations should normally
-	be 0, 1, ... in this order. In case of doubt, the translations for each case should match the value
-	returned by the formula for Plural-Forms. */
-	std::string ns = ngettext("0", "1", number);
-	unsigned int n = 0;
-	static bool warned = false;
-	if (ns.length() == 1 && '0' <= ns[0] && ns[0] <= '9') {
-		n = ns[0] - '0';
-	} else if (!warned) {
-		warned = true;
-		errorstream << "Translations: plurals are not translated correctly in the current language; got \""
-		            << ns << "\" instead of the integer selecting which plural to use" << std::endl;
+	std::wstring key = textdomain + L"|" + s;
+	auto it = m_plural_translations.find(key);
+	if (it != m_plural_translations.end()) {
+		auto n = (*(it->second.first))(number);
```

Comment:
We probably should check here that `n` is between `0` and some arbitrary limit like `10` to avoid a badly-formed translation file to use an arbitrary amount of memory on the client.

Reply:
`n` is limited by the size of `it->second.second` (i.e. the array of plural translations) a few lines below.

Reply:
Ah yes, thanks. I was confused and I thought we still grew `it->second.second` if smaller than `n`.